src/test/java/org/jenkinsci/plugins/workflow/log/FileLogStorageTest.java

```diff
@@ -55,6 +57,48 @@ public class FileLogStorageTest extends LogStorageTestBase {
         assertOverallLog(0, lines("stuff"), true);
     }
 
+    @Test public void corruptIndex() throws Exception {
+        FileUtils.writeStringToFile(log, "before\n1\nbetween1\n2\nbetween2\n3\nafter", StandardCharsets.UTF_8);
+        FileUtils.writeStringToFile(new File(log + "-index"), "7 1\n?\n18 2\n20\n29 3\n31", StandardCharsets.UTF_8);
```

Comment:
Multi-line strings might be more legible here, though you would need to careful about placement of newlines and indentation.